src/tools/recipe2plan.ts

```diff
@@ -0,0 +1,162 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+import {Runtime} from '../runtime/runtime.js';
+import {Manifest} from '../runtime/manifest.js';
+import {IsValidOptions, Recipe, RecipeComponent} from '../runtime/recipe/recipe.js';
+import {CapabilitiesResolver, StorageKeyOptions} from '../runtime/capabilities-resolver.js';
+import {RecipeResolver} from '../runtime/recipe/recipe-resolver.js';
+import {Arc} from '../runtime/arc.js';
+import {Loader} from '../platform/loader-web.js';
+import {Store} from '../runtime/storageNG/store.js';
+import {Exists} from '../runtime/storageNG/drivers/driver.js';
+import {TestVolatileMemoryProvider} from '../runtime/testing/test-volatile-memory-provider.js';
+import {RamDiskStorageDriverProvider, RamDiskStorageKey} from '../runtime/storageNG/drivers/ramdisk.js';
+import {Capabilities} from '../runtime/capabilities.js';
+import {ramDiskStorageKeyPrefixForTest, storageKeyPrefixForTest} from '../runtime/testing/handle-for-test.js';
+
+
+/**
+ * Generates Kotlin Plans from recipes in an arcs manifest.
+ *
+ * @param path path/to/manifest.arcs
+ * @return Generated Kotlin code.
+ */
+export async function recipe2plan(path: string): Promise<string> {
+  const manifest = await Runtime.parseFile(path);
+
+  const recipes = new StorageKeyRecipeResolver(manifest).resolve();
+
+  const plans = await generatePlans(recipes);
+
+  return plans.join('\n');
+}
+
+
+/**
+ * Converts each resolved recipes into a Kotlin Plan class.
+ *
+ * @param resolutions A series of resolved recipes.
+ * @return List of generated Kotlin plans
+ */
+async function generatePlans(resolutions: AsyncIterator<Recipe>): Promise<string[]> {
+  // TODO Implement
+  return [''];
+}
+
+
+/**
+ * Responsible for resolving recipes with storage keys.
+ */
+export class StorageKeyRecipeResolver {
```

Comment:
Might be worth putting this in its own file, and writing specific unit tests for it